frontend-react/src/config/endpoints/lookupTables.ts

```diff
@@ -0,0 +1,63 @@
+import { HTTPMethods, RSEndpoint } from ".";
+
+// the shape used by the frontend client for value sets
+export interface ValueSet {
+    name: string;
+    createdBy: string;
+    createdAt: string;
+    system: string;
+}
+
+// the shape sent down by the API for value sets
+export interface ApiValueSet {
+    name: string;
+    created_by: string; // unused
+    created_at: string; // unused
+    system: string;
+    reference: string; // unused
+    referenceURL: string; // unused
+}
+
+export interface ValueSetRow {
+    name: string;
+    code: string;
+    display: string;
+    version: string;
+}
+
+export interface LookupTable {
+    lookupTableVersionId: number;
+    tableName: string;
+    tableVersion: number;
+    isActive: boolean;
+    createdBy: string;
+    createdAt: string;
+    tableSha256Checksum: string;
+}
+
+export enum LookupTables {
+    VALUE_SET = "sender_automation_value_set",
+    VALUE_SET_ROW = "sender_automation_value_set_row",
+}
+
+// notice the react-router style colon demarcated dynamic path segments
+export const lookupTablesEndpoints = {
```

Comment:
Is there a plan to build up some typing around these API configurations? Ensuring anything exported from any endpoint file is an object where values can only be RSEndpoints helps keep us within our pattern.

Reply:
It looks like it can be achieved simply with:
```typescript
interface ApiOrWhatever {
    [key: string]: RSEndpoint
}
```
That _should_ leave things still `.notation` accessible even though we stringify the key.

Reply:
I've added a new type RSApiEndpoints that describes what I feel like the export from an endpoints file will be. Beyond that, I get what you're saying. I can envision implementing something like, well, a resource, I guess, that would be instantiated with an RSApiEndpoints object. We could then set up a standardized set of fetch functions for each endpoint on the resource, and expose those to our hooks that are making requests. One of the benefits of this is that it could organize things in a way that would nicely mirror a RESTful API, and ensure that the returned values from the API match exactly what we'd expect. This is similar to the road I went down on this branch at first, but given the points I brought up earlier today (the non-RESTful nature of our API, the ambiguity around return values from the API in some cases, the ability to type things pretty well without bringing resources into the picture), decided against it. Maybe there is another way to get what you're thinking of into the app (if you can think of one definitely let me know), but I think what's here - a standard fetch function provided to all hooks that expects a configuration of a specific shape, coupled with the ability to house relevant variables and functionality within a standardized class format - allows us pretty good safety while still being flexible.

Reply:
@doug-s-nava Actually, I think we're on the same page about resources and not tying the endpoints to them -- I was just advocating for this type so as an engineer working on a new Api integration, I can rely on the type to give me linter/compiler errors like "hey, that's not an RSEndpoint, and it needs to be"! 👍🏼